File: frontend/app/src/main/java/com/dailymemo/domain/models/Memo.kt
package com.dailymemo.domain.models

import java.time.LocalDateTime

data class Memo(
    val id: Long,
    val userId: Long,
    val title: String,
    val content: String, // Deprecated - will be removed
    val creationMode: CreationMode = CreationMode.LIST, // NEW: 생성 방식 (map/list)
    val categories: List<MemoCategory> = emptyList(), // NEW: 선택된 카테고리 목록
    val imageUrl: String?,
    val imageUrls: List<String> = emptyList(), // 여러 이미지 지원
    val rating: Float, // 0.0 ~ 5.0 (0.5 단위)
    val isPinned: Boolean,
    val likesCount: Int = 0, // 좋아요 개수
    val isLiked: Boolean = false, // 현재 사용자의 좋아요 여부
    val latitude: Double?,
    val longitude: Double?,
    val locationName: String?,
    val category: PlaceCategory = PlaceCategory.OTHER, // Deprecated
    val isWishlist: Boolean = false, // 위시리스트 여부 (true=가고싶은곳, false=방문한곳)
    val businessName: String? = null, // 장소/가게명
    val businessPhone: String? = null, // 전화번호
    val businessAddress: String? = null, // 주소
    val naverPlaceUrl: String? = null, // 네이버 플레이스 URL
    val comments: List<Comment> = emptyList(), // 댓글 목록
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)

data class MemoListResult(
    val memos: List<Memo>,
    val total: Long,
    val page: Int,
    val limit: Int,
    val hasMore: Boolean
)

File: backend/src/features/memo/model/response/memo.go
package response

import (
	commentResponse "main/features/comment/model/response"
	"time"
)

// ResCategory 카테고리 응답 구조체
type ResCategory struct {
	ID           uint   `json:"id"`
	Name         string `json:"name"`
	Sentiment    string `json:"sentiment"`
	Color        string `json:"color"`
	DisplayOrder int    `json:"display_order"`
}

type ResMemo struct {
	ID              uint                         `json:"id"`
	UserID          uint                         `json:"user_id"`
	Title           string                       `json:"title"`
	Content         string                       `json:"content"` // Deprecated - will be empty after migration
	CreationMode    string                       `json:"creation_mode"` // NEW: map or list
	Categories      []ResCategory                `json:"categories"` // NEW: 선택된 카테고리 배열
	ImageURL        string                       `json:"image_url"`
	Rating          float32                      `json:"rating"` // Changed from uint8 to float32 for decimal support
	IsPinned        bool                         `json:"is_pinned"`
	LikesCount      uint                         `json:"likes_count"`
	IsLiked         bool                         `json:"is_liked"`
	Latitude        *float64                     `json:"latitude"`
	Longitude       *float64                     `json:"longitude"`
	LocationName    *string                      `json:"location_name"`
	Category        *string                      `json:"category"` // Deprecated
	// Wishlist fields (Issue #19)
	IsWishlist      bool                         `json:"is_wishlist"`
	BusinessName    *string                      `json:"business_name,omitempty"`
	BusinessPhone   *string                      `json:"business_phone,omitempty"`
	BusinessAddress *string                      `json:"business_address,omitempty"`
	NaverPlaceURL   *string                      `json:"naver_place_url,omitempty"`
	Comments        []commentResponse.ResComment `json:"comments,omitempty"`
	CreatedAt       time.Time                    `json:"created_at"`
	UpdatedAt       time.Time                    `json:"updated_at"`
}

type ResMemoList struct {
	Memos []ResMemo `json:"memos"`
	Total int64     `json:"total"`
	Page  int       `json:"page"`
	Limit int       `json:"limit"`
}
